apps/spruce/src/pages/waterfall/useFilters.ts

```diff
@@ -64,14 +72,37 @@ export const useFilters = (waterfall: WaterfallQuery["waterfall"]) => {
       ),
     [versions],
   );
+  const buildVariantFilterRegex: RegExp[] = useMemo(
+    () =>
+      buildVariantFilter.reduce<RegExp[]>((accum, curr) => {
+        let variantRegex;
+        try {
+          variantRegex = new RegExp(curr, "i");
+        } catch {
+          return accum;
+        }
+        return [...accum, variantRegex];
+      }, []),
+    [buildVariantFilter],
+  );
 
   const buildVariants = useMemo(() => {
     if (!hasFilters) {
       return waterfall.buildVariants;
     }
 
     const bvs: WaterfallBuildVariant[] = [];
-    waterfall.buildVariants.forEach((bv) => {
+    const waterfallBuildVariants = buildVariantFilterRegex.length
+      ? waterfall.buildVariants.filter(({ displayName }) => {
```

Comment:
Could we apply this filter within the forEach? As we build out filtering I think it would be preferable to minimize the number of mapping operations